File: examples/python-examples/PointsAnalyze.dox.py
import meshlib.mrmeshpy as mrmeshpy

# Analyze a text-based point cloud file before loading it
analysis_result = mrmeshpy.PointsLoad.analyzeText("pointcloud.csv")

# Print analysis results
print("File Analysis Results:")
print(f"  Total lines: {analysis_result.totalLines}")
print(f"  Data lines: {analysis_result.dataLines}")
print(f"  Comment lines: {analysis_result.commentLines}")
print(f"  Empty lines: {analysis_result.emptyLines}")
print(f"  Has normals: {'yes' if analysis_result.hasNormals else 'no'}")
print(f"  Has colors: {'yes' if analysis_result.hasColors else 'no'}")

if analysis_result.header:
    print(f"  Header: {analysis_result.header}")

if analysis_result.firstDataLine:
    print(f"  First data line: {analysis_result.firstDataLine}")

# Now load the full point cloud
point_cloud = mrmeshpy.loadPoints("pointcloud.csv")

print(f"Loaded {point_cloud.validPoints.count()} points")

File: examples/cpp-examples/PointsAnalyze.dox.cpp
#include <MRMesh/MRPointsLoad.h>
#include <MRMesh/MRPointCloud.h>

#include <iostream>

int main()
{
    // Analyze a text-based point cloud file before loading it
    auto analysisResult = MR::PointsLoad::analyzeText( "pointcloud.csv" );
    
    if ( !analysisResult )
    {
        std::cerr << "Failed to analyze file: " << analysisResult.error() << std::endl;
        return 1;
    }
    
    // Print analysis results
    std::cout << "File Analysis Results:" << std::endl;
    std::cout << "  Total lines: " << analysisResult->totalLines << std::endl;
    std::cout << "  Data lines: " << analysisResult->dataLines << std::endl;
    std::cout << "  Comment lines: " << analysisResult->commentLines << std::endl;
    std::cout << "  Empty lines: " << analysisResult->emptyLines << std::endl;
    std::cout << "  Has normals: " << ( analysisResult->hasNormals ? "yes" : "no" ) << std::endl;
    std::cout << "  Has colors: " << ( analysisResult->hasColors ? "yes" : "no" ) << std::endl;
    
    if ( !analysisResult->header.empty() )
        std::cout << "  Header: " << analysisResult->header << std::endl;
    
    if ( !analysisResult->firstDataLine.empty() )
        std::cout << "  First data line: " << analysisResult->firstDataLine << std::endl;
    
    // Now load the full point cloud
    auto pointCloud = MR::PointsLoad::fromText( "pointcloud.csv" );
    
    if ( !pointCloud )
    {
        std::cerr << "Failed to load point cloud: " << pointCloud.error() << std::endl;
        return 1;
    }
    
    std::cout << "Loaded " << pointCloud->validPoints.count() << " points" << std::endl;
    
    return 0;
}
